src/httpserver.cpp

```diff
@@ -459,14 +477,22 @@ void StopHTTPServer()
         evhttp_del_accept_socket(eventHTTP, socket);
     }
     boundSockets.clear();
+    {
+        WAIT_LOCK(g_requests_mutex, lock);
+        while (!g_requests.empty()) {
+            g_requests_cv.wait(lock);
+        }
+    }
+    if (eventHTTP) {
+        event_base_once(eventBase, -1, EV_TIMEOUT, [](evutil_socket_t, short, void*) {
```

Comment:
maybe include a comment explaining this, see https://github.com/bitcoin/bitcoin/pull/19420#discussion_r833476744

Reply:
done, I hope that covers it.